src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/ConfigureSwaggerGeneratorOptions.cs

```diff
@@ -34,17 +35,57 @@ public void Configure(SwaggerGeneratorOptions options)
 
             // Create and add any filters that were specified through the FilterDescriptor lists ...
 
-            _swaggerGenOptions.ParameterFilterDescriptors.ForEach(
-                filterDescriptor => options.ParameterFilters.Add(GetOrCreateFilter<IParameterFilter>(filterDescriptor)));
+            foreach(var filterDescriptor in _swaggerGenOptions.ParameterFilterDescriptors)
+            {
+                if (filterDescriptor.Type.IsAssignableTo(typeof(IParameterFilter)))
+                {
+                    options.ParameterFilters.Add(GetOrCreateFilter<IParameterFilter>(filterDescriptor));
+                }
+
+                if (filterDescriptor.Type.IsAssignableTo(typeof(IParameterAsyncFilter)))
+                {
+                    options.ParameterAsyncFilters.Add(GetOrCreateFilter<IParameterAsyncFilter>(filterDescriptor));
+                }
```

Comment:
If a class implements both interfaces, won't this mean it gets registered as both kinds and runs twice? Should it do that, or should it prefer one interface over the other?

Reply:
As a developer if I would implement both I would expect both to be run.

Reply:
But would you expect it to be instantiated twice, once for each interface? I think it would make more sense to instantiate it, then put the same instance into one or both sets.

Reply:
Should this only apply to filters added as instances or also for filters added as types with di? Because registering the same filter twice with e.g. `options.OperationFilter<T>` already creates two different instances, so I would also keep this behaviour even if the class is the same. When using e.g. `options.AddOperationFilterInstance(TFilter instance)` I also agree that only one instance for both implementations should be added.

Reply:
Same as if you add a class that implements both a schema filter and an operation filter (non async variants). It would instantiate it twice. I think it's less surprising to keep that behavior